### PR TITLE
Holoclean v.0.1.1

### PR DESCRIPTION
**Pull request by gmichalo**
In the pull request from dev to master we will add:
Features:
  1) k-inference values
  2) unittests for pruning and initfeaturizer
  3) adding reset_database() method that deletes all the tables from the database
  4) updating reader to check for columns with values with more than 250 characters
drops these columns for now and notifies user.
  5) updating parser_interface to check if we have more than one attribute per predicate
  raise exception early to prevent errors in error detection

Cleanup/Bug Fixes:
  1) Removing unused variables from softmax.py
  2) Fixing method remove_denial_constraint in holoclean.py
  3) Removing __ind collumn from the final table
  4) Updating the python_requirement.txt - removing unused packages
  5) Updating README.md - dropping pytorch install as pip handles it and some virtualenv instructions

### CODE REVIEW

File: python-package-requirement.txt [hunk 1]
```diff
@@ -1,65 +1,15 @@
...
 torchvision==0.2.0
-tornado==5.0.1
 tqdm==4.20.0
-traitlets==4.3.2
-wcwidth==0.1.7
-webencodings==0.5.1
-widgetsnbextension==3.2.0
+scipy==1.0.1
+Pandas==0.20.3
+
```
Comment (thodrek) [inline python-package-requirement.txt]: why an empty + ?

**Review by thodrek — CHANGES_REQUESTED** [review]
Everything looks good to me apart from the evaluation print outs

"The MAP recall is : 0.978378378378 out of 185"

We should round to three decimals (%.3f) and also you are missing something " what 0.9 out of 185" even mean. Please fix that.

**Comment by thodrek** [comment]
Oh another thing that needs to be fixed. When I run with different FDs I get a different number of items (i.e., not 185 in hospital). This is wrong. The set cells we evaluate on should be those that have a different value in groundtruth vs init.

File: holoclean/holoclean.py [hunk 2]
```diff
@@ -553,9 +569,8 @@ def _add_error_detector(self, new_error_detector):
         :return: Null
 
         """
-        self.holo_env.logger.info('getting the  for error detection...')
         self.error_detectors.append(new_error_detector)
-        self.holo_env.logger.info('getting new for error detection')
+        self.holo_env.logger.info('Added new error detection')
```
Comment (jw-mcgrath) [inline holoclean/holoclean.py:573]: "Added new error detector"

**Review by jw-mcgrath — COMMENTED** [review]
Great work guys! Only thing I noticed was this, but all the code looks good.

**Review by jw-mcgrath — APPROVED** [review]

**Review by thodrek — APPROVED** [review]
